chromadb/segment/__init__.py

```diff
@@ -0,0 +1,43 @@
+from abc import ABC, abstractmethod
+from collections.abc import Sequence
+from typing import TypedDict, Union
+import numpy.typing as npt
+
+IDSeq = Sequence[str]
+Vector = Sequence[Union[int, float]]
+
+
+class Query(TypedDict):
+    """A KNN/ANN query to the segment"""
+    vector: Vector
+    k: int = 10
+    approximate: bool = True
```

Comment:
I don't know if we want to expose this functionality to users for resource-isolation in a multi-tenant environment. One query could hog a disproportionate amount of resources by saturating us with brute force queries on larger collections. The sublinear scaling of ANN indices gives us more workable bounds.

Reply:
Aren't there other index types that support exact search without doing a full brute force?

Anyway my point in putting this here wasn't that we'd allow any type of user to make any type of query; just that, for segment types that allow query-time parameters, this is the appropriate place to put them.

You're right, we probably shouldn't expose this in the user-facing API and should default to the normal behavior based on the segment type.

Reply:
I guess things like a sorted list fall in that category. That sgtm.

Reply:
Should query time parameters be less concretely represented? For example the HNSW index exposes several query parameters, guessing those should be passed as a typed dict particular to that segment type.

Reply:
Indeed. I'll change to a dict named "parameters".